frontend/mina-signer/src/MinaSigner.ts

```diff
@@ -11,11 +11,35 @@ import {
   Payment,
   StakeDelegation,
   Message,
+  Party,
+  SignableData,
 } from "./TSTypes";
 
+import { isPayment, isMessage, isStakeDelegation, isParty } from "./Utils";
+
 const defaultValidUntil = "4294967295";
 
-class Client {
+// A variable to check if we have cleaned up any wasm workers ran by jsoo
+let didShutdown: Boolean = false;
+
+/**
+ * Checks the 'didShtudown' variable and performs a clean up on any
+ * wasm workers that were ran by the jsoo. This should only be called when
+ * the signer is no longer used. Intended for node environments, this is a noop
+ * for browsers.
+ */
+// @ts-ignore
+export function shutdown() {
```

Comment:
Instead of exporting this function, we should execute its logic right here at the top level